src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -126,7 +126,8 @@ public long getLongMetric(String metric)
       return ((Number) metricValue).longValue();
     } else if (metricValue instanceof String) {
       try {
-        return Long.valueOf(((String) metricValue).replace(",", ""));
+        String s = ((String) metricValue).replace(",", "");
+        return s.contains(".") ? Double.valueOf(s).longValue() : Long.valueOf(s);
```

Comment:
we should also handle the + sign, otherwise "+9223372036854770000" will return 9223372036854769664L instead of 9223372036854770000